### PR TITLE
fix(FR-3499): stop the image selector clearing itself on re-select

### LINKED ISSUE #8667: Image environment selector: clicking an option does not select the image (Session Launcher, Deployment revision)
## Component

- **Name**: the image Environment / Version selector — `ImageEnvironmentSelectFormItems` (`react/src/components/ImageEnvironmentSelectFormItems.tsx`), which renders two `BAISelect` instances (`packages/backend.ai-ui/src/components/BAISelect.tsx`)
- **Astryx primitive**: `Select` (via `BAISelect`, children option API with `BAISelectOptionItem` / `BAISelectOptionGroup`)
- **Scope**: every usage — the reporter reproduced it on both the Session Launcher and the Deployment revision flow

## Where it is reachable

- Session Launcher — the "Environments / Version" step (menu: Sessions → Start)
- Deployment revision — `DeploymentAddRevisionModal.tsx`, the image selection field
- Also used in: 4 non-test files import `ImageEnvironmentSelectFormItems` or its `Image` type

(`SessionLauncherPage.tsx`, `DeploymentAddRevisionModal.tsx`,

`SessionFormItems/ResourceAllocationFormItems.tsx`, `hooks/useDefaultImagesWithFallback.ts`).

This is usage breadth from a grep, not a per-usage reproduction.

## Steps to reproduce

1. Open the Session Launcher (Sessions → Start) — or the Deployment revision modal.
1. Click the **Environments** selector; the option list opens.
1. Click any image option in the list.
1. Repeat on the **Version** selector.

## Observed

The image cannot be selected at all:

- clicking an option in the list has no effect — the option does not become

selected and the list stays as it was; and

- in the cases where a click does register, the chosen image is not reflected

back into the form field (the selector / form keeps its previous value).

Both the Environments and the Version selector are affected, and it happens on

the Session Launcher and the Deployment revision flow alike, so no session or

deployment can be created through the normal image-selection path.

## Expected

Clicking an option selects it: the dropdown closes, the trigger shows the chosen

environment/image, and the form value (`environments.environment`,

`environments.version`, `environments.image`) is updated so the Session Launcher

and the Deployment revision modal can proceed to the next step.

## Environment

- **Mode**: Unknown (not reported; not mode-specific as far as reported)
- **Viewport**: Unknown
- **Browser/OS**: Unknown
- **Build**: Unknown — reported on the current Astryx UI

## Impact

Blocks the task — image selection is a required step for both session creation

and deployment revision creation, and no workaround was reported. The reporter

flagged it as urgent.

## Evidence

- Screenshot / recording: none supplied.

## Notes

Reported in Korean; both symptom variants above are the reporter's own wording

("항목 클릭이 안 먹힘", "선택은 되지만 값이 반영 안 됨"). Recorded as one issue because

they were reported as one defect on the same control; split later if triage finds

two mechanisms.

JIRA Issue: FR-3499

### PR DESCRIPTION
**Pull request by agatha197**
Resolves #8667 (FR-3499)

## Summary

The image environment selector cleared its own value when the user clicked an option, on the Session Launcher and in the deployment **Add Revision** modal alike.

**Root cause.** Every option in the environment select is keyed by `environmentGroup.environmentName` — `registry/namespace` — but its `onChange` wrote back the image's bare `namespace`:

```tsx
environment: (supportExtendedImageInfo ? firstInListImage?.namespace : firstInListImage?.name) || '',
```

Nothing downstream matches that shorter form. The version select resolves its group by `environmentName`, and Astryx `Selector` renders its **placeholder** when no option carries the current value — so the environment trigger went blank and the version select emptied with it.

**Why it looked intermittent.** Choosing a *different* environment masked the bug: the resulting `version` change re-ran the auto-select effect, which rewrote the field with the correct `environmentName`. Choosing the environment that was **already selected** changes no version, so the effect never ran and the broken value survived. On a cluster with a single installed environment that is *every* click — which is why it presents as "이미지를 아예 선택할 수 없다".

The fix writes the option's own key back, and looks the group up with `_.find` so a miss can no longer throw on `[0].images[0]`.

## Also fixed — the trigger printed its own search key

Found while measuring the above. `BAISelect` folded an option's `filterValue` into the `label` it hands Astryx, but `Selector` spends `SelectorOptionData.label` **twice**: the dropdown filters on it *and* the closed trigger renders it (`selectedItem?.label`). The synthetic search key was therefore painted onto the control:

| | before | after |
|---|---|---|
| Environments trigger | `Python Python` | `Python` |
| Version trigger | `3.9 x86_64 3.9 Ubuntu 20.04 x86_64` | `3.9 x86_64` |
| Version trigger (multi-image cluster) | `2.1.0 x86_64 2.1.0⇥Python 3.10⇥x86_64⇥GPU⇥CUDA12.1⇥Ubuntu 22.04` | `2.1.0 x86_64 Python 3.10 GPU:CUDA12.1 Ubuntu 22.04` |

The label is now the display text alone. Search filters on the text the user is reading — which is exactly what this module's existing `optionFilterProp` PILOT-DECISION already specifies for the `options`-prop path, so the two option APIs now agree instead of only one of them corrupting itself.

**Trade-off worth a reviewer's eye:** extras that appeared only in `filterValue` and not in the visible row (e.g. an environment's `prefix` tag) are no longer searchable. Astryx's option model has no separate search field, so a clean trigger and an extended search key are mutually exclusive here.

## Verification

- `bash scripts/verify.sh` → `=== ALL PASS ===` (Relay / Lint / Format / TypeScript / Vite warmup / StyleX / Astryx theme build / Terminology)
- `pnpm --filter backend.ai-ui exec vitest run` → 586 passed, 1 skipped
- `pnpm --prefix ./react exec vitest run` → 1162 passed
- `pnpm --filter backend.ai-ui build` → OK
- New regression test `BAISelect.triggerLabel.test.tsx` (3 cases). Confirmed it **fails without** the fix with the exact string measured live: `expected 'PyTorch PyTorch\tstable\tGPU' to be 'PyTorch'`.
- **Live probe** against a real backend, Session Launcher + deployment Add-Revision modal: clicking the already-selected environment used to blank both selects (`Select…` / `Select…`); it now keeps `Python` / `3.9 x86_64`. Selecting a different environment (ngc-vllm → PyTorch → TensorFlow) still updates both. No new `pageerror`.

**Checklist:**

- [ ] Documentation
- [ ] Minimum required manager version
- [ ] Specific setting for review
- [x] Minimum requirements to check during review: re-select the environment that is already selected — the value must survive; and confirm the search trade-off above is acceptable
- [x] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

File: packages/backend.ai-ui/src/components/BAISelect.tsx [hunk 1]
```diff
@@ -290,9 +290,11 @@ function BAISelect<ValueType = any, OptionType = BAISelectOption>({
   // Astryx's `Selector` is `options`-driven only, so the element tree is
   // flattened into the same option model — the group's `label` becomes a
   // `{type: 'section'}` entry (Astryx's native grouping), each option's JSX
-  // body is kept for `renderOption`, and `filterValue` (the synthetic search
-  // key `ImageEnvironmentSelectFormItems` sets) is folded into the searchable
-  // label. Two call sites use this form; both nest options inside OptGroups.
+  // body is kept for `renderOption`, and the option's flattened text becomes
+  // its label. `filterValue` (the synthetic search key
+  // `ImageEnvironmentSelectFormItems` sets) is NOT folded in — see FR-3499 at
+  // the leaf branch below. Two call sites use this form; both nest options
```
[the remark this answers is not in the document]
Comment (agatha197) [reply packages/backend.ai-ui/src/components/BAISelect.tsx:296]: Good catch — fixed in 9cc3c77eb. The JSDoc now states that `filterValue` is accepted and ignored, and that search matches the option's visible text only, with a pointer to why (Astryx renders the option label on the trigger too, so anything folded into it leaks there).

The prop itself is kept in the signature so the antd-era `optionFilterProp="filterValue"` call sites keep compiling.

**Review by nowgnuesLee — APPROVED** [review]
LGTM
